heron/spi/src/java/com/twitter/heron/spi/statefulstorage/Checkpoint.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.spi.statefulstorage;
+
+import com.twitter.heron.proto.ckptmgr.CheckpointManager;
+
+public class Checkpoint {
+  private final String topologyName;
+  private final String checkpointId;
+  private final String componentName;
+  private final String instanceId;
+  private final String taskId;
+
+  private CheckpointManager.SaveInstanceStateRequest saveBytes;
```

Comment:
I'll rename the saveBytes, but this description should go in the proto file actually when we do make changes to a more structured form